pkg/auth/rhsso_authenticator.go

```diff
@@ -176,8 +189,13 @@ func (a *RHSSOAuthenticator) AuthUserAuth(token string) (interface{}, error) {
 
 	// Check if the parsed token is valid...
 	if !parsedToken.Valid {
-		a.log.Errorf("Token is invalid: %s", parsedToken.Raw)
-		return nil, errors.Errorf("Token is invalid: %s", parsedToken.Raw)
+		// Don't report error "Token used before issued"
+		// TODO: This validation is going to be removed in jwt-go v4, once jwt-go v4
+		// is released and we start using it, this validation-skip can be removed.
+		if !isValidationErrorIssuedAt(err) {
```

Comment:
This error couldn't have changed from the last call, right?
I think this second check is a waste.

Reply:
So I have tried without it at first but the new unit-test failed.
Apparently, they are 2 places that we can fail on (the second is `parsedToken.Valid`) and if we don't skip the second one as well we will fail on a different error (but for the same reason).

I don't know why it was implemented that way from the first place though.

Reply:
Hmm, but are we allowing an otherwise invalid token here if the issued at time is just wrong?

Reply:
We should think about this a bit more. What we're saying here is that we have an error from parsing that says that the issued at time is incorrect which we want to skip. But what else does `Valid` check?

If there are other checks, then we're ignoring those for tokens that have "used before issued" errors, right?

Reply:
That may be a good point.

Let me check `jwt-go` code that handles that and I will write here about what I have discovered.

Reply:
OK, I have fix the error check to make sure we get only the `issuedAt` error and no other errors as well.
```
$ git diff
diff --git a/pkg/auth/rhsso_authenticator.go b/pkg/auth/rhsso_authenticator.go
index d796723f..cecb6f6c 100644
--- a/pkg/auth/rhsso_authenticator.go
+++ b/pkg/auth/rhsso_authenticator.go
@@ -156,7 +156,7 @@ func isValidationErrorIssuedAt(err error) bool {
        if !ok {
                return false
        }
-       return e.Errors&jwt.ValidationErrorIssuedAt != 0
+       return e.Errors == jwt.ValidationErrorIssuedAt
 }
```

Reply:
I will wait for you `/unhold`